financialaid/views.py

```diff
@@ -224,23 +227,56 @@ class FinancialAidActionView(UpdateAPIView):
     queryset = FinancialAid.objects.all()
 
 
-class GetLearnerPriceForCourseView(APIView):
+class CoursePriceListView(APIView):
     """
-    View for retrieving a leaner's price for a course run
+    View for retrieving a learner's price for course runs in all enrolled programs
+    """
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        """
+        GET handler
+        """
+        user = request.user
+        program_enrollments = ProgramEnrollment.objects\
+            .select_related('user', 'program')\
+            .filter(user=user, program__live=True).all()
+        formatted_course_prices = []
+        for program_enrollment in program_enrollments:
+            try:
+                response_dict = get_formatted_course_price(program_enrollment)
+                response_dict['program_id'] = program_enrollment.program.id
+                formatted_course_prices.append(response_dict)
+            except ImproperlyConfigured:
```

Comment:
Returning an empty list isn't really more helpful for the user. If we do get that exception it is something we need to deal with immediately, else the financial aid and ecommerce parts of the app will fail when the user makes use of them. By raising the `ImproperlyConfigured` exception we are saying that the configuration is broken and needs to be repaired by the administrator, the application is not able to be useful to the user (especially not the course list API part of it)